ouroboros/shared/value.go

```diff
@@ -72,7 +72,7 @@ func (v Value) AddAsset(coins ...Coin) {
 		if _, ok := v[coin.assetId.PolicyID()]; !ok {
 			v[coin.assetId.PolicyID()] = map[string]num.Int{}
 		}
-		v[coin.assetId.PolicyID()][coin.assetId.AssetName()] = coin.amount
+		v[coin.assetId.PolicyID()][coin.assetId.AssetName()] = v[coin.assetId.PolicyID()][coin.assetId.AssetName()].Add(coin.amount)
```

Comment:
did you test this? I know num.Int behaves weirdly with these methods.

Reply:
I did locally. I'll add a unit test to make it official.